packages/react-sdk/src/resources-generator.test.ts

```diff
@@ -0,0 +1,142 @@
+import { expect, test } from "@jest/globals";
+import stripIndent from "strip-indent";
+import { createScope, type Page } from "@webstudio-is/sdk";
+import { generateResourcesLoader } from "./resources-generator";
+
+const clear = (input: string) =>
+  stripIndent(input).trimStart().replace(/ +$/, "");
+
+const toMap = <T extends { id: string }>(list: T[]) =>
+  new Map(list.map((item) => [item.id, item] as const));
+
+test("generate getjson resource loader", () => {
+  expect(
+    generateResourcesLoader({
+      scope: createScope(),
+      page: { rootInstanceId: "body" } as Page,
+      dataSources: toMap([
+        {
+          id: "variableResourceId",
+          scopeInstanceId: "body",
+          type: "resource",
+          name: "resourceName",
+          resourceId: "resourceId",
+        },
+      ]),
+      resources: toMap([
+        {
+          id: "resourceId",
+          instanceId: "body",
+          type: "getjson",
+          url: `"https://my-json.com"`,
+        },
+      ]),
+    })
+  ).toEqual(
+    clear(`
+      const _getjson = async (url: string) => {
+        const response = await fetch(url)
+        if (response.ok) {
+          const data = await response.json()
+          return data;
+        }
+        return {
+          error: await response.text()
+        }
+      }
+      type Params = Record<string, string | undefined>
+      export const loadResources = async (_props: { params: Params }) => {
+      const [
```

Comment:
I know its only a test, but is there an easy way to make it look nicely formatted? Also why not use inline snapshots?

Reply:
> I know its only a test, but is there an easy way to make it look nicely formatted?

Formatting makes code generation code more complex. Pieces of code are quite small so not completely unreadable.

> Also why not use inline snapshots?
Because jest doesn't work with prettier v3

Reply:
so we are not using inline snapshots at all now? I would wait for prettier support and not upgrade, inline snapshots are important we should be able to use them

Reply:
We use prettier v3 for a long time. Always better to bump it earlier to avoid massive git conflicts. Inline snapshots don't do much honestly. And output has escaped double quotes which are harder to read.